app/controllers/api/generic_works_controller.rb

```diff
@@ -0,0 +1,98 @@
+module API
+  # Adds an endpoint that customizes the update action on nested works
+  class GenericWorksController < ApplicationController
```

Comment:
Would a name change do? Something like WorkRelationshipsController, or are you describing a deeper change? Maybe generating this controller based on some rake task that generates `Image` types of works and the supporting files?